pkg/netutil/netutil.go

```diff
@@ -103,7 +104,26 @@ func resolveURL(ctx context.Context, lg *zap.Logger, u url.URL) (string, error)
 		)
 		return "", err
 	}
-	if host == "localhost" || net.ParseIP(host) != nil {
+	if net.ParseIP(host) != nil {
+		p, err := strconv.Atoi(u.Port())
+		if err != nil {
+			lg.Warn(
+				"failed to parse URL Port while resolving URL",
+				zap.String("url", u.String()),
+				zap.String("host", u.Host),
+				zap.String("port", u.Port()),
+				zap.Error(err),
+			)
+			return "", err
+
+		}
+		addr := &net.TCPAddr{
+			IP:   net.ParseIP(host),
+			Port: p,
+		}
+		return addr.String(), nil
+	}
```

Comment:
Probably we can just remove this code block, so that it always execute `resolveTCPAddr` (line 130)

Reply:
Thanks for the feedback. I've removed that block as suggested and amended the overridden resolveTCPAddr functions in the other tests in this package to handle valid IP addresses.